src/components/SmartCollapse/SmartCollapse.tsx

```diff
@@ -0,0 +1,83 @@
+import ChevronRightOutlined from '@mui/icons-material/ChevronRightOutlined';
+import { Box, Button, Stack, useTheme } from '@mui/joy';
+import { ReactNode, useEffect, useState } from 'react';
+
+import { useUserPreferences } from '@/contexts/UserPreferencesContext';
+
+import { SmartCollapseKey, smartCollapses } from './smartCollapses';
+
+export type SmartCollapseProps = {
+  children: ReactNode;
+  /**
+   * Height must be manually set for transitions to work properly
+   */
+  approxHeightPx: number;
+  /**
+   * If you want the collapsed state to be persisted to local storage,
+   * pass an id for this section
+   */
+  id?: SmartCollapseKey;
+};
+
+export function SmartCollapse({
+  children,
+  approxHeightPx,
+  id,
+}: SmartCollapseProps) {
+  const initiallyCollapsed =
+    smartCollapses.find((x) => x.key === id)?.isDefaultCollapsed ?? false;
+  const [collapsed, setCollapsed] = useState(initiallyCollapsed);
+  const {
+    userPrefs: { collapsedSections },
+    updateUserPrefs,
+  } = useUserPreferences();
+
+  useEffect(() => {
+    if (id) {
+      setCollapsed(collapsedSections[id]);
+    }
+  }, [collapsedSections, id]);
+
+  const toggle = () => {
+    setCollapsed((collapsed) => !collapsed);
+    if (id) {
+      updateUserPrefs({
+        collapsedSections: {
+          // DANGER! This value can be stale, need to fix this.
```

Comment:
@eshedmargalit , thoughts on this? I'm worried that with multiple of these on screen, they'll overwrite each other. The solution is probably to create a special function which only updates this id's collapse?